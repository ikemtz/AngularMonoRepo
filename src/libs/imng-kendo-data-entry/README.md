# imng-kendo-data-entry

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `ng test imng-kendo-data-entry` to execute the unit tests.
