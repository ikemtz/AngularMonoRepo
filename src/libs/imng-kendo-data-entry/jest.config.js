module.exports = {
  preset: '../../jest.preset.js',
  setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
  globals: {
    'ts-jest': {
      stringifyContentPathRegex: '\\.(html|svg)$',

      tsconfig: '<rootDir>/tsconfig.spec.json',
    },
  },
  coverageDirectory: '../../coverage/libs/imng-kendo-data-entry',

  displayName: 'imng-kendo-data-entry',
  snapshotSerializers: [
    'jest-preset-angular/build/serializers/no-ng-attributes',
    'jest-preset-angular/build/serializers/ng-snapshot',
    'jest-preset-angular/build/serializers/html-comment',
  ],
  transform: { '^.+\\.(ts|js|html)$': 'jest-preset-angular' },
  reporters: [
    'default',
    [
      'jest-junit',
      {
        outputDirectory: '.',
        outputName: './junit.kendo-data-entry.xml',
      },
    ],
  ],
};
