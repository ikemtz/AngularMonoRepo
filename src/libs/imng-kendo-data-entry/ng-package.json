{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../dist/imng-kendo-data-entry",
  "lib": {
    "entryFile": "src/public-api.ts"
  }
}
