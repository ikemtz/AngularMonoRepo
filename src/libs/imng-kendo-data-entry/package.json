{
  "name": "imng-kendo-data-entry",
  "description": "A library to make integrating Angular Kendo forms and NGRX a bit simpler.",
  "version": "0.0.1",
  "peerDependencies": {
    "@angular/common": ">= 14",
    "@angular/core": ">= 14",
    "@angular/forms": ">= 14",
    "@angular/platform-browser": ">= 14",
    "@ngrx/store": ">= 14",
    "@progress/kendo-angular-dialog": ">= 6",
    "imng-ngrx-utils": "*",
    "imng-nrsrx-client-utils": "*",
    "rxjs": "*"
  },
  "dependencies": {
    "tslib": "^2.3.0"
  },
  "keywords": [
    "Angular",
    "Kendo",
    "Forms",
    "NGRX"
  ],
  "license": "MIT",
  "repository": {
    "type": "git",
    "url": "https://github.com/ikemtz/AngularMonoRepo.git"
  },
  "icon": "https://avatars.githubusercontent.com/u/6444182?v=4",
  "private": false,
  "author": {
    "name": "Isaac Martinez <@ikemtz>",
    "url": "https://github.com/ikemtz"
  }
}
