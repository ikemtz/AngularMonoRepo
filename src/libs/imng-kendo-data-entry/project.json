{
  "projectType": "library",
  "sourceRoot": "libs/imng-kendo-data-entry/src",
  "prefix": "imng",
  "targets": {
    "build": {
      "executor": "@nrwl/angular:package",
      "outputs": ["dist/libs/imng-kendo-data-entry"],
      "options": {
        "project": "libs/imng-kendo-data-entry/ng-package.json"
      },
      "dependsOn": [
        {
          "target": "build",
          "projects": "dependencies"
        }
      ],
      "configurations": {
        "production": {
          "tsConfig": "libs/imng-kendo-data-entry/tsconfig.lib.prod.json"
        },
        "development": {
          "tsConfig": "libs/imng-kendo-data-entry/tsconfig.lib.json"
        }
      },
      "defaultConfiguration": "production"
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/imng-kendo-data-entry"],
      "options": {
        "jestConfig": "libs/imng-kendo-data-entry/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/imng-kendo-data-entry/src/**/*.ts",
          "libs/imng-kendo-data-entry/src/**/*.html",
          "libs/imng-kendo-data-entry/testing/**/*.ts",
          "libs/imng-kendo-data-entry/testing/**/*.html"
        ]
      }
    }
  },
  "tags": []
}
