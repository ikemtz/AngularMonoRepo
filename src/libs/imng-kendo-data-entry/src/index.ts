export * from './lib/data-entry.module';
export * from './lib/data-entry-dialog.component';
export * from './lib/base-data-entry.component';
export * from './lib/data-entry-facade';
export * from './lib/reducer-collection-logic';
