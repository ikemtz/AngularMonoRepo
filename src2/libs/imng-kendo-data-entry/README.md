# imng-kendo-data-entry

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test imng-kendo-data-entry` to execute the unit tests.
