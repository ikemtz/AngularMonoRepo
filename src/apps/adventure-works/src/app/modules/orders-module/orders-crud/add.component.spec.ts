import { CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA } from '@angular/core';
import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { DatePickerModule } from '@progress/kendo-angular-dateinputs';
import { DropDownsModule } from '@progress/kendo-angular-dropdowns';
import { createDataEntryMockFacade } from 'imng-kendo-data-entry/testing';
import {
  mockConsoleError,
  mockConsoleGroup,
  mockConsoleWarn,
  readFirst,
} from 'imng-ngrx-utils/testing';
import { of } from 'rxjs';
import {
  IOrder,
  createTestOrder,
  createTestCustomer,
  createTestSalesAgent,
  createTestOrderAddress,
} from '../../../models/odata';

import { OrderAddComponent } from './add.component';
import { OrderCrudFacade } from './crud.facade';

export function createMockOrderFacade() {
  return {
    currentEntity$: of({}),
    customers$: of([
      {
        id: 'abc',
        num: 'abc',
        name: 'abc',
        companyName: 'abc',
        emailAddress: 'abc',
        phone: 'abc',
      },
      {
        id: 'xyz',
        num: 'xyz',
        name: 'xyz',
        companyName: 'xyz',
        emailAddress: 'xyz',
        phone: 'xyz',
      },
    ]),
    loadCustomers: jest.fn(),
    shipToAddresses$: of([
      {
        id: 'abc',
        line1: 'abc',
        line2: 'abc',
        city: 'abc',
        stateProvince: 'abc',
        countryRegion: 'abc',
        postalCode: 'abc',
      },
      {
        id: 'xyz',
        line1: 'xyz',
        line2: 'xyz',
        city: 'xyz',
        stateProvince: 'xyz',
        countryRegion: 'xyz',
        postalCode: 'xyz',
      },
    ]),
    loadShipToAddresses: jest.fn(),
    billToAddresses$: of([
      {
        id: 'abc',
        line1: 'abc',
        line2: 'abc',
        city: 'abc',
        stateProvince: 'abc',
        countryRegion: 'abc',
        postalCode: 'abc',
      },
      {
        id: 'xyz',
        line1: 'xyz',
        line2: 'xyz',
        city: 'xyz',
        stateProvince: 'xyz',
        countryRegion: 'xyz',
        postalCode: 'xyz',
      },
    ]),
    loadBillToAddresses: jest.fn(),
  };
}

describe('OrderAddComponent', () => {
  let component: OrderAddComponent;
  let fixture: ComponentFixture<OrderAddComponent>;
  let facade: OrderCrudFacade;
  let consoleWarnMock: jest.SpyInstance<void>;
  let consoleGroupMock: jest.SpyInstance<void>;

  beforeEach(waitForAsync(() => {
    consoleWarnMock = mockConsoleWarn();
    consoleGroupMock = mockConsoleGroup();
    TestBed.configureTestingModule({
      declarations: [OrderAddComponent],
      imports: [
        ReactiveFormsModule,
        NoopAnimationsModule,
        DatePickerModule,
        DropDownsModule,
      ],
      providers: [
        {
          provide: OrderCrudFacade,
          useValue: createDataEntryMockFacade(createMockOrderFacade()),
        },
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(OrderAddComponent);
    component = fixture.componentInstance;
    facade = TestBed.inject(OrderCrudFacade);
    fixture.detectChanges();
  });

  afterAll(() => {
    component.ngOnDestroy();
    consoleWarnMock.mockRestore();
    consoleGroupMock.mockRestore();
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should save', () => {
    component.initForm();
    component.addEditForm.patchValue(createTestOrder());
    component.addEditForm.controls.billToAddress?.patchValue(
      createTestOrderAddress(),
    );
    component.addEditForm.controls.shipToAddress?.patchValue(
      createTestOrderAddress(),
    );
    component.addEditForm.controls.customer?.patchValue(createTestCustomer());
    component.addEditForm.controls.customer?.controls.salesAgent?.patchValue(
      createTestSalesAgent(),
    );
    let item: IOrder | undefined;
    facade.saveNewEntity = jest.fn((x) => (item = x));
    facade.updateExistingEntity = jest.fn();
    expect(component.getFormErrors()).toStrictEqual([]);
    component.onSubmit();
    expect(facade.saveNewEntity).toHaveBeenCalledTimes(1);
    expect(facade.updateExistingEntity).toHaveBeenCalledTimes(0);

    expect(item).toMatchSnapshot({
      date: expect.any(Date),
      dueDate: expect.any(Date),
      shipDate: expect.any(Date),
    });
  });

  /**
   * Note: if this test fails, then you're missing validators in your forms.
   * Using form validators is typically a good idea.
   */
  test('should not save', () => {
    const consoleErrorMock = mockConsoleError();
    component.addEditForm?.patchValue({});
    component.onSubmit();
    expect(facade.saveNewEntity).toHaveBeenCalledTimes(0);
    expect(facade.updateExistingEntity).toHaveBeenCalledTimes(0);
    consoleErrorMock.mockRestore();
  });

  test('should cancel', () => {
    facade.clearCurrentEntity = jest.fn();
    component.cancel();
    expect(facade.clearCurrentEntity).toHaveBeenCalledTimes(1);
  });

  test('should support Customer filters', async () => {
    component.handleCustomerFilter('xy');
    const result = await readFirst(component.customers$);
    expect(result).toStrictEqual([
      {
        id: 'xyz',
        num: 'xyz',
        name: 'xyz',
        companyName: 'xyz',
        emailAddress: 'xyz',
        phone: 'xyz',
      },
    ]);
  });

  test('should support ShipToAddress filters', async () => {
    component.handleShipToAddressFilter('xy');
    const result = await readFirst(component.shipToAddresses$);
    expect(result).toStrictEqual([
      {
        id: 'xyz',
        line1: 'xyz',
        line2: 'xyz',
        city: 'xyz',
        stateProvince: 'xyz',
        countryRegion: 'xyz',
        postalCode: 'xyz',
      },
    ]);
  });

  test('should support BillToAddress filters', async () => {
    component.handleBillToAddressFilter('xy');
    const result = await readFirst(component.billToAddresses$);
    expect(result).toStrictEqual([
      {
        id: 'xyz',
        line1: 'xyz',
        line2: 'xyz',
        city: 'xyz',
        stateProvince: 'xyz',
        countryRegion: 'xyz',
        postalCode: 'xyz',
      },
    ]);
  });
  test('should handle StatusType filters', async () => {
    component.handleStatusTypeFilter('abc-xyz');
    const result = await readFirst(component.statusTypes$);
    expect(result).toStrictEqual([]);
  });
  test('should handle ShippingType filters', async () => {
    component.handleShippingTypeFilter('abc-xyz');
    const result = await readFirst(component.shippingTypes$);
    expect(result).toStrictEqual([]);
  });
});
