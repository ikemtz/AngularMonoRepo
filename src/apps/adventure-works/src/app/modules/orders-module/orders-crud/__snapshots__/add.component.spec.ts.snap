// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderAddComponent should save 1`] = `
Object {
  "billToAddress": "BILL_TO_ADDRESS",
  "billToAddressId": "BILL_TO_ADDRESS_ID",
  "comment": "COMMENT",
  "creditCardApprovalCode": "CREDIT_CARD_APP",
  "customer": "CUSTOMER",
  "customerId": "CUSTOMER_ID",
  "date": Any<Date>,
  "dueDate": Any<Date>,
  "freight": 0,
  "id": undefined,
  "isOnlineOrder": true,
  "num": "NUM",
  "orderId": 0,
  "orderLineItems": null,
  "purchaseOrderNum": "PURCHASE_ORDER_NUM",
  "revisionNum": 0,
  "shipDate": Any<Date>,
  "shipMethod": "SHIP_METHOD",
  "shipToAddress": "SHIP_TO_ADDRESS",
  "shipToAddressId": "SHIP_TO_ADDRESS_ID",
  "status": 0,
  "subTotal": 0,
  "taxAmt": 0,
  "totalDue": 0,
}
`;
